api/app/controllers/mno_enterprise/impersonate_controller.rb

```diff
@@ -15,7 +15,11 @@ def create
       else
         flash[:notice] = "User doesn't exist"
       end
-      redirect_to mnoe_home_path
+
+      path = mnoe_home_path
+      path = add_param_to_fragment(path + "#!/" , 'dhbRefId', params[:dhbRefId]) if params[:dhbRefId].present?
```

Comment:
I'm not sure you need the `+ "!/"` see the [add_param_to_fragment](https://github.com/maestrano/mno-enterprise/blob/d3cddef17bf068e6ef14a20cb79fdcd0e6423d97/core/spec/controllers/mno_enterprise/application_controller_spec.rb#L15-L20)